rb/spec/integration/selenium/webdriver/manager_spec.rb

```diff
@@ -53,103 +53,221 @@ module WebDriver
-          driver.manage.add_cookie name: 'samesite', value: 'strict', same_site: 'Strict'
+        it 'should add secure cookie when https',
+           exclusive: {driver: :none,
+                       reason: "Can only be tested on https site"} do
+          driver.get 'https://www.selenium.dev'
 
-          expect(driver.manage.cookie_named('samesite')[:same_site]).to eq('Strict')
+          driver.manage.add_cookie name: 'secure',
+                                   value: 'https',
+                                   secure: true
+
+          expect(driver.manage.cookie_named('secure')[:secure]).to eq(true)
         end
 
-        it 'should add sameSite cookie with attribute Lax', only: {browser: %i[chrome edge firefox]} do
-          driver.navigate.to url_for('xhtmlTest.html')
-          driver.manage.add_cookie name: 'samesite',
-                                   value: 'lax',
-                                   same_site: 'Lax'
-          expect(driver.manage.cookie_named('samesite')[:same_site]).to eq('Lax')
+        context 'sameSite' do
+          it 'should add with attribute Strict', only: {browser: %i[chrome edge firefox]} do
+            driver.manage.add_cookie name: 'samesite',
+                                     value: 'strict',
+                                     same_site: 'Strict'
+
+            expect(driver.manage.cookie_named('samesite')[:same_site]).to eq('Strict')
+          end
+
+          it 'should add with attribute Lax', only: {browser: %i[chrome edge firefox]} do
+            driver.manage.add_cookie name: 'samesite',
+                                     value: 'lax',
+                                     same_site: 'Lax'
+            expect(driver.manage.cookie_named('samesite')[:same_site]).to eq('Lax')
+          end
+
+          it 'should add with attribute None',
+             exclusive: {driver: :none,
+                         reason: "Can only be tested on https site"} do
+            driver.get 'https://selenium.dev'
+
+            driver.manage.add_cookie name: 'samesite',
+                                     value: 'none-secure',
+                                     same_site: 'None',
+                                     secure: true
+
+            expect(driver.manage.cookie_named('samesite')[:same_site]).to eq('None')
+          end
+
+          it 'should not be added with attribute None when not secure',
+             except: {browser: :firefox,
+                      reason: "https://github.com/mozilla/geckodriver/issues/1842"} do
+            expect {
+              driver.manage.add_cookie name: 'samesite',
+                                       value: 'none-insecure',
+                                       same_site: 'None',
+                                       secure: false
+            }.to raise_exception(Error::UnableToSetCookieError)
+          end
+        end
+
+        context 'expiration' do
+          it 'should add with DateTime' do
```

Comment:
for all of these how about?

`it can be a DateTime`
`can be a Time`
`can be a Float` or Number?
`cannot be any value in the past`

Reply:
as far as naming? Yeah, ideally we get rid of all of the "shoulds," I try to keep it consistent within the file at least and didn't feel like changing the others. :)

Reply:
oki, then we could change to `should be a DateTime` e.t.c. I think the "add with" is confusing here.

Reply:
You have to read the whole thing: 
> cookie management expiration should add with DateTime

better: 
> cookie management expiration should allow adding with DateTime value

ideally:
> cookie management expiration respects DateTime value

Reply:
No I know that, just that first sentence sounds bad imo. Doesn't read well in english. Other 2 are fine.